skrf/vi/vna/vna.py

```diff
@@ -0,0 +1,739 @@
+
+    def stop_freq(self, channel: int = 1) -> float:
+        """
+        Get stop frequency
+
+        Parameters
+        ----------
+        channel : int
+            channel to get frequency (if supported)
+
+        Returns
+        -------
+        float
+            stop frequency [Hz]
+        """
+        return float(self.query(f"sense{channel}:frequency:stop?"))
+
+    def set_stop_freq(self, f: float, channel: int = 1) -> None:
+        """
+        Set stop frequency
+
+        Parameters
+        ----------
+        f : float
+            stop frequency [Hz]
+        channel : int
+            channel to set frequency (if supported)
+        """
+        self.write(f"sense{channel}:frequency:stop {f}")
+
+    def npoints(self, channel: int = 1) -> int:
+        """
+        Get number of frequency points
+
+        Parameters
+        ----------
+        channel : int
+            channel to get npoints (if supported)
+
+        Returns
+        -------
+        int
+            number of frequency points
+        """
+        return int(self.query(f"sense{channel}:sweep:points?"))
+
+    def set_npoints(self, n: int, channel: int = 1) -> None:
+        """
+        Set number of frequency points
+
+        Parameters
+        ----------
+        n : int
+            number of frequency points
+        channel : int
+            channel to set npoints (if supported)
+        """
+        self.write(f"sense{channel}:sweep:points {n}")
+
+    def freq_step(self, channel: int = 1) -> float:
+        """
+        Get frequency step
+
+        Parameters
+        ----------
+        channel : int
+            channel to get frequency step (if supported)
+
+        Returns
+        -------
+        float
+            frequency step [Hz]
+        """
+        return float(self.query(f"sense{channel}:sweep:step?"))
+
+    def set_freq_step(self, f: float, channel: int = 1) -> None:
+        """
+        Set frequency step
+
+        Parameters
+        ----------
+        f : float
+            frequency step [Hz]
+        channel : int
+            channel to set frequency step (if supported)
+        """
+        self.write(f"sense{channel}:sweep:step {f}")
+
+    def frequency(self, channel: int = 1) -> Frequency:
+        """
+        Get current frequency as :class:`Frequency` object
+
+        Parameters
+        ----------
+        channel : int
+            channel to get frequency (if supported)
+
+        Returns
+        -------
+        Frequency
+            current frequency settings as frequency object
+        """
+        start = self.start_freq(channel)
+        stop = self.stop_freq(channel)
+        npoints = self.npoints(channel)
+        return Frequency(start, stop, npoints, unit="hz")
+
+    def set_frequency(
+        self,
+        frequency: Optional[Frequency] = None,
+        start: Optional[float] = None,
+        stop: Optional[float] = None,
+        npoints: Optional[int] = None,
```

Comment:
could it be that it takes as input a `Frequency` object for consistency with the package?

Reply:
That absolutely makes more sense.. idk why I had it this. Will make the change

Reply:
[6cffc50](https://github.com/scikit-rf/scikit-rf/pull/744/commits/6cffc5040dfaccd9e90533533a2f0d911005115a)